numpy/core/tests/test_dtype.py

```diff
@@ -1240,6 +1243,8 @@ class dt:
         assert np.dtype(dt()) == np.float64
 
     def test_recursion(self):
+        if hasattr(sys, "pyston_version_info"):
```

Comment:
Why switch from the decorator?

Reply:
Changed to decorators for all but `test_to_bool_scalar()`, since only the self reference part fails while the first part of the test passes. 
Do I need to do a new pull request?